packages/request-manager/src/controller.ts

```diff
@@ -12,23 +12,32 @@ export class TorController extends EventEmitter {
     waitingTime = 1000;
     maxTriesWaiting = 60;
     isCircuitEstablished = false;
+    torIsDisabledWhileStarting: boolean;
 
     constructor(options: TorConnectionOptions) {
         super();
         this.options = options;
+        this.torIsDisabledWhileStarting = false;
         this.controlPort = new TorControlPort(options, this.onMessageReceived.bind(this));
     }
 
     onMessageReceived(message: string) {
         const bootstrap: BootstrapEvent[] = bootstrapParser(message);
-        this.isCircuitEstablished = bootstrap.some(
-            (event: BootstrapEvent) => event.progress === BootstrapEventProgress.Done,
-        );
+        bootstrap.forEach(event => {
+            if (!event || !event.progress) return;
```

Comment:
note: I might find an answer myself later, so this is more like a note - there could be events which don't have progress property and we want to ignore them?

nitpick: `!event?.progress` maybe nicer?

Reply:
Yes, that is waaaay nicer. 4584a907ca6928379596cf6a58dceed3cb13dec6